src/main/java/net/imglib2/Dimensions.java

```diff
@@ -61,4 +64,42 @@ default void dimensions( final long[] dimensions )
 	 * @param d
 	 */
 	public long dimension( int d );
+
+	/**
+	 * Check that all entries in dimensions are positive
+	 *
+	 * @param dimensions
+	 * @return true if all entries in dimension are positive, false otherwise
+	 */
+	public static boolean dimensionsCheckAllPositive( final long[] dimensions )
```

Comment:
I will rename to`allPositive`, it reads better.

I am conflicted about `long...`. I had used it initially, but it is only useful when you check dimensions by hand, and I don't think that something like this is very common:
```
Dimensions.allPositive(1, 2, 3, ....);
ArrayImgs.longs(1, 2, 3, ...);
```
Instead of typing the dimensions twice, I would rather imagine something like this:
```
final long[] dims = {1, 2, 3, ...};
Dimesnions.allPositive(dims);
ArrayImgs.longs(dims);
```

Maybe this could be addressed by changing the return type to `long[]` and returning the argument if the check passes (`null` otherwise or throw exception).